### PR TITLE
Use ``argparse`` config handler on three checkers

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

## Description

### CODE REVIEW

**Comment by DanielNoord** [comment]
Blocked by #5921.

**Review by Pierre-Sassoulas — APPROVED** [review]
Let's do the migration from python 3.6 now then.

**Comment by DanielNoord** [comment]
> Let's do the migration from python 3.6 now then.

We're still merging hot fixes. Still is a fairly trivial PR. I would be okay with waiting a bit. There is enough else for me to do!

**Comment by Pierre-Sassoulas** [comment]
OK, after https://github.com/PyCQA/pylint/milestone/61 is closed then ?

**Comment by DanielNoord** [comment]
> OK, after https://github.com/PyCQA/pylint/milestone/61 is closed then ?

I'd say, let's wait a minimal week since last hotfix release. That gives people ample time to report any new issues. I agree with Marc, the change to `3.7` is likely going to make releasing hotfixes much more difficult.

**Comment by Pierre-Sassoulas** [comment]
> last hotfix release. 

We continuously release false positive fixes now so they won't be a last hotfix release. Do you mean the last one with crashes fixed inside it ?

>  releasing hotfixes much more difficult.

I agree too, but the question is when do we release 2.14 and not ever hot fix python 3.6 anymore :) Depends on how blocked we are. There are maybe 4 MR now and 2.14 is close to 100 issues already (lots of documentation but still). So, I'm already considering releasing 2.14.

**Comment by DanielNoord** [comment]
> We continuously release false positive fixes now so they won't be a last hotfix release. Do you mean the last one with crashes fixed inside it ?

Yeah, or false positives introduced by `2.12`.

> I agree too, but the question is when do we release 2.14 and not ever hot fix python 3.6 anymore :) Depends on how blocked we are. There are maybe 4 MR now and 2.14 is close to 100 issues already (lots of documentation but still). So, I'm already considering releasing 2.14.

If possible I would wait with `2.14` until the `argparse` migration has been finished. That way we can do a "light" release with a move to `3.7` + a request for plugin maintainers on feedback on the new options handling. Looking at the speed at which it's going now I think that it is feasible to do so.
Only 227 references to `.config` remain in the codebase right now.

File: pylint/checkers/base/name_checker/checker.py
Comment (Pierre-Sassoulas) [inline pylint/checkers/base/name_checker/checker.py:305]: Should we handle this in the option parsing directly and not later with getattr ? I think it's not ideal as it's a delayed default value that put a little of option parsing inside checkers.

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

File: pylint/checkers/base/name_checker/checker.py
Comment (DanielNoord) [reply pylint/checkers/base/name_checker/checker.py:305]: These options are created dynamically for each name type we have.

We can probably define them in `checker.options` but that's more a general refactor of this checker. By creating them dynamically we do remove a lot of duplication of the code though.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by DanielNoord** [comment]
Missing coverage is also missing on `main` so I'm merging.

**Pull request merged**
